From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Horst <horst.git@protonmail.com>
Date: Thu, 4 Mar 2021 11:56:54 +0000
Subject: [PATCH] Bug 1685883 - building with --disable-marionette fails with
 compile error r=keeler,whimboo,smaug

building firefox 84.0.2 from source (https://archive.mozilla.org/pub/firefox/releases/84.0.2/source/) fails when using `ac_add_options --disable-marionette` because of missing header

Differential Revision: https://phabricator.services.mozilla.com/D101390

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 dom/base/Navigator.cpp                         | 14 +++++++++++++-
 security/manager/ssl/nsCertOverrideService.cpp |  6 +++++-
 toolkit/moz.configure                          |  4 ++--
 3 files changed, 20 insertions(+), 4 deletions(-)

diff --git a/dom/base/Navigator.cpp b/dom/base/Navigator.cpp
index aa3fa6719f05..0fc1db890519 100644
--- a/dom/base/Navigator.cpp
+++ b/dom/base/Navigator.cpp
@@ -70,6 +70,9 @@
 #include "nsICookieManager.h"
 #include "nsICookieService.h"
 #include "nsIHttpChannel.h"
+#ifdef ENABLE_MARIONETTE
+#  include "nsIMarionette.h"
+#endif
 #include "nsStreamUtils.h"
 #include "WidgetUtils.h"
 #include "nsIScriptError.h"
@@ -2105,7 +2108,16 @@ webgpu::Instance* Navigator::Gpu() {
 
 /* static */
 bool Navigator::Webdriver() {
-  return Preferences::GetBool("marionette.enabled", false);
+  bool marionetteRunning = false;
+
+#ifdef ENABLE_MARIONETTE
+  nsCOMPtr<nsIMarionette> marionette = do_GetService(NS_MARIONETTE_CONTRACTID);
+  if (marionette) {
+    marionette->GetRunning(&marionetteRunning);
+  }
+#endif
+
+  return marionetteRunning;
 }
 
 }  // namespace dom
diff --git a/security/manager/ssl/nsCertOverrideService.cpp b/security/manager/ssl/nsCertOverrideService.cpp
index d6b9e8f606b4..9ee9776224ae 100644
--- a/security/manager/ssl/nsCertOverrideService.cpp
+++ b/security/manager/ssl/nsCertOverrideService.cpp
@@ -16,7 +16,9 @@
 #include "nsAppDirectoryServiceDefs.h"
 #include "nsCRT.h"
 #include "nsILineInputStream.h"
-#include "nsIMarionette.h"
+#ifdef ENABLE_MARIONETTE
+#  include "nsIMarionette.h"
+#endif
 #include "nsIObserver.h"
 #include "nsIObserverService.h"
 #include "nsIOutputStream.h"
@@ -615,10 +617,12 @@ static bool IsDebugger() {
   bool marionetteRunning = false;
   bool remoteAgentListening = false;
 
+#ifdef ENABLE_MARIONETTE
   nsCOMPtr<nsIMarionette> marionette = do_GetService(NS_MARIONETTE_CONTRACTID);
   if (marionette) {
     marionette->GetRunning(&marionetteRunning);
   }
+#endif
 
 #ifdef ENABLE_REMOTE_AGENT
   nsCOMPtr<nsIRemoteAgent> agent = do_GetService(NS_REMOTEAGENT_CONTRACTID);
diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 8ae672508dac..d4fc3dc6769e 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -961,8 +961,8 @@ def marionette(value):
     if value:
         return True
 
-set_config('ENABLE_MARIONETTE', marionette)
-
+set_config("ENABLE_MARIONETTE", marionette)
+set_define("ENABLE_MARIONETTE", marionette)
 
 # geckodriver WebDriver implementation
 # ==============================================================
-- 
2.31.1

